relayer/pkg/starknet/client.go

```diff
@@ -49,18 +49,18 @@ type ReaderWriter interface {
 
 var _ ReaderWriter = (*Client)(nil)
 
-var _ caigotypes.Provider = (*Client)(nil)
+// var _ caigotypes.Provider = (*Client)(nil)
```

Comment:
if this doesn't need to conform to the `Provider` type anymore, we can remove the `Unimplemented` interface + related functions